### PR TITLE
Ensures closing tabs works reliably and as expected

### PR DESCRIPTION
**Pull request by sorsaffari**
## What is the goal of this PR?
Changes introduced in this PR, ensure that closing a tab always works and it does not reset the state of other tabs.

## What are the changes implemented in this PR?
resolves https://github.com/graknlabs/workbase/issues/176
resolves https://github.com/graknlabs/workbase/issues/288

### `VisualiserContent.vue`: Use and Object instead of a Map for `tabs`
Maps are not reactive in Vue (yet).
Objects have a simpler syntax compared to Maps, which provide better readability

### VisualiserContent.vue`: Remove extra logic from `closeTab`
The following is weird (!) and fortunately no longer needed
```
// re-set the same tab to trigger dynamic rerendering of list
const temp = this.currentTab;	
this.currentTab = null;	
this.currentTab = temp;
```

### Add e2e tests for interacting with Tabs

### CODE REVIEW

**Comment by flyingsilverfin** [comment]
I'm not exactly sure if I can review this without a lot more knowledge of our system? What do you think, should i just review for style?

**Comment by flyingsilverfin** [comment]
Im' out of the JS eco system - a map and an Object are different??

File: src/renderer/components/Visualiser/VisualiserContent.vue [hunk 1]
```diff
@@ -134,34 +133,31 @@ export default {
...
-        const temp = this.currentTab;
-        this.currentTab = null;
-        this.currentTab = temp;
-      }
+      const visTabToClose = this.$children.find(x => x.tabId && x.tabId === tab);
+      visTabToClose.$destroy();
+
+      // vue doesn't detect property deleteion, so we clone, delete and reassign
+      const tabsClone = { ...this.tabs };
+      delete tabsClone[tab];
```
Comment (flyingsilverfin) [inline src/renderer/components/Visualiser/VisualiserContent.vue:147]: this doesn't end up re-drawing the tab does it?

**Comment by sorsaffari** [comment]
> Im' out of the JS eco system - a map and an Object are different??

a `Map` is an `Object` (like pretty much everything else in Javascript) that holds key-value pairs. It has its own API that's designed for managing key-value pairs. a `Map` is exactly what we'd need here for holding tab id-name pairs. However, Vue can't (yet) watch for changes made to a map to re-render (update) the component. this is why I'm replacing `Map` with a raw object (i.e. `tabs`) which we structure ourselves to hold key-value pairs (tabs' id-name).

**Comment by sorsaffari** [comment]
> I'm not exactly sure if I can review this without a lot more knowledge of our system? What do you think, should i just review for style?

I don't blame you. I think for this PR in particular, the unit tests could be reviewed for completion. but again you aren't a hardcore Workbase user either 🤷‍♂️

File: src/renderer/components/Visualiser/VisualiserContent.vue (hunk 1 above)
Comment (sorsaffari) [reply src/renderer/components/Visualiser/VisualiserContent.vue:147]: no, it wouldn't. to understand why this cloning is required, one would have to understand the Vue reactivity in-depth. that's why I've left a comment on line 145 to explain the reason for cloning.

Comment (flyingsilverfin) [reply src/renderer/components/Visualiser/VisualiserContent.vue:147]: fair enough!

**Pull request merged**